include/realtime_tools/realtime_publisher.hpp

```diff
@@ -49,173 +49,336 @@
 
 namespace realtime_tools
 {
-template <class Msg>
-class RealtimePublisher
+
+// Allow to select the way the RT Publisher is implemented
+enum class RealtimePublisherPollingMode { Polling, NonPolling };
+
+// Provide backwards compatibility for NON_POLLING define
+#ifndef NON_POLLING
+#define NON_POLLING RealtimePublisherPollingMode::Polling
+#else
+#undef NON_POLLING
+#define NON_POLLING RealtimePublisherPollingMode::NonPolling
+// Note this is the only way to somehow deprecate a define as #warning is not portable
```

Comment:
Why not doing something like here?
https://github.com/ros-controls/realtime_tools/blob/3ee84c57c2530820224b25f7b28f50f3901f4bf8/include/realtime_tools/realtime_box.h#L20-L27

Reply:
We can also go for that way. I assume it works for all supported platforms.